src/main/java/com/mykaarma/urlshortener/service/UrlService.java

```diff
@@ -136,20 +146,34 @@ public UrlDetails shortenUrl(String longUrl, String shortUrlDomain, long expiryD
 		}
 
 		String shortUrl = shortUrlDomain + urlPrefix + shortUrlHash;
-		
+
 		shortUrlDetails.setShortUrl(shortUrl);
-		
+
 		try {
 			urlRepository.saveUrl(shortUrlDetails);
+		}
+		catch (ShortUrlDuplicateException e)
+		{
+			log.info(String.format("Mongo Exception Occurred, retrying creating short url for longUrl=%s businessUUID=%s domainPurpose=%s",longUrl, businessUUID, shortUrlDomain));
```

Comment:
Don't use word `mongo` in logs, use something like DuplicateHashKeyException occurred, retrying ....